src/description.ts

```diff
@@ -1,21 +1,91 @@
-import { DEFAULT_LOCALE_DESCRIPTION } from './constants'
-import { Descriptions, LocalizedJSONSchema } from './types'
+import { JSONSchema6, JSONSchema6Definition } from 'json-schema'
+import { DEFAULT_DESCRIPTION_LOCALE } from './constants'
+import { Descriptions, LocalizedJSONSchema, LocalizedJSONSchemaDefinition } from './types'
 import { isNotEmpty } from './utils'
 
-export function getDescriptions (schema: LocalizedJSONSchema, localeDescription: string = DEFAULT_LOCALE_DESCRIPTION): Descriptions {
+export function getDescriptions (schema: LocalizedJSONSchema, descriptionLocale: string = DEFAULT_DESCRIPTION_LOCALE): Descriptions {
   const description = isNotEmpty(schema.description) ? schema.description : undefined
   const descriptions = schema.descriptions
 
   if (descriptions == null) {
-    return description != null ? { [localeDescription]: description } : {}
+    return description != null ? { [descriptionLocale]: description } : {}
   }
 
-  if (description != null && !Object.keys(descriptions).includes(localeDescription)) {
-    descriptions[localeDescription] = description
+  if (description != null && !Object.keys(descriptions).includes(descriptionLocale)) {
+    descriptions[descriptionLocale] = description
   }
   return descriptions
 }
 
-export function getLocalizedDescription (schema: LocalizedJSONSchema, locale: string, localeDescription?: string): string | undefined {
-  return getDescriptions(schema, localeDescription)[locale]
+export function getLocalizedDescription (schema: LocalizedJSONSchema, locale: string, descriptionLocale?: string): string | undefined {
+  return getDescriptions(schema, descriptionLocale)[locale]
+}
+
+function renderSimpleJsonSchema (locale: string, schema?: LocalizedJSONSchemaDefinition, descriptionLocale?: string): JSONSchema6Definition | undefined {
+  return schema == null
+    ? undefined
+    : typeof schema === 'boolean'
+      ? schema
+      : renderJsonSchema(schema, locale, descriptionLocale)
+}
+
+function renderArrayOfJsonSchema (locale: string, array?: LocalizedJSONSchemaDefinition[], descriptionLocale?: string): JSONSchema6Definition[] | undefined {
+  return array?.map(schema => renderSimpleJsonSchema(locale, schema, descriptionLocale) as JSONSchema6Definition)
+}
+
+function renderObjectOfJsonSchema (locale: string, object?: { [key: string]: LocalizedJSONSchemaDefinition }, descriptionLocale?: string): { [key: string]: JSONSchema6Definition } | undefined {
+  return object == null
+    ? undefined
+    : Object.entries(object).reduce((newSchema, [key, schema]) => ({
+      ...newSchema,
+      [key]: renderSimpleJsonSchema(locale, schema, descriptionLocale)
+    }), {})
+}
+
+export function renderJsonSchema (schema: LocalizedJSONSchema, locale: string, descriptionLocale?: string): JSONSchema6 {
```

Comment:
What is the difference between `locale` and `descriptionLocale`?

Reply:
`locale` is the locale of the description we want to retrieve and `descriptionLocale` is the locale of the default `description` in the JSON schema.

For example, if we have a question only in french, the `locale` will be `'fr'` to retrieve the french description but `descriptionLocale` is `'en'` for our questions.

Reply:
d0c7e4f4433ebee9dfbce2c2e3b09c15e5273e6a